File: src/week1/credit/credit.c
#include "..\..\..\lib\cs50.h"
// #include <cs50.h>
#include <stdio.h>
#include <stdbool.h>

#include <stdlib.h>

string judge(const int answer, const int length);
int main(void)
{
    // printf("%lld\n",sizeof(long long));
    long long num = get_long_long("Number: ");
    int sum = 0, temp = 0, length = 0, ans = 0;
    bool isEvenPlace = false;

    for (size_t i = 0; num != 0; i++)
    {

        temp = isEvenPlace ? (num % 10) * 2 : num % 10;
        sum += temp >= 10 ? temp / 10 + temp % 10 : temp;
        isEvenPlace = !isEvenPlace;
        num /= 10;
        if (num >= 10 && num < 100)
        {
            ans = num;
        }

        length++;
    }
    // printf("ans:%d,len:%d\n", sum, length);
    printf("%s\n", sum % 10 == 0 ? judge(ans, length) : "INVALID");
    return 0;
}

string judge(const int answer, const int length)
{
    if ((answer == 34 || answer == 37) && length == 15)
    {
        return "AMEX";
    }
    else if (answer >= 51 && answer <= 55 && length == 16)
    {
        return "MASTERCARD";
    }
    else if (answer >= 40 && answer <= 49 && (length == 13 || length == 16))
    {
        return "VISA";
    }
    else
    {
        return "INVALID";
    }
}

File: src/week6/sentimental-credit/credit.py
# TODO
import sys

num = input("Number: ")


temp, sum, length = 0, 0, len(str(num))

if length != 13 and length != 15 and length != 16:
    print("INVALID")
    sys.exit()

isEvenPlace = True if length % 2 == 0 else False

for i in range(length):
    if isEvenPlace:
        temp = int(num[i])*2
        sum += (temp - 9) if temp >= 10 else temp
        isEvenPlace = False
    else:
        sum += int(num[i])
        isEvenPlace = True

answer = int(num[0])*10+int(num[1])

if sum % 10 == 0:
    if answer == 34 or answer == 37:
        print("AMEX")
    elif 51 <= answer <= 55:
        print("MASTERCARD")
    elif int(num[0]) == 4:
        print("VISA")
    else:
        print("INVALID")
else:
    print("INVALID")
